### PR TITLE
Fix #9772 - SONAR: code quality for 13 release (ImageCropperRenderer)

### LINKED ISSUE #9772: SONAR: code quality for 13 release
### Description

The codebase might need some attention since last time we focused on this, on this day:
- 131 critical
- 87 major

More details here: https://sonarcloud.io/project/issues?resolved=false&id=org.primefaces%3Aprimefaces

I see a lot of `Make "XXX" transient or serializable"` kind of errors, and I think it'd be great to know once and for all what to do with these warnings?

The rest should be quite easy to fix. If you have any questions, suggestions, feel free to share 😉  

### Describe the solution you would like

See how we can split the work (it's redundant work and not so fun 😅 )

### Additional context

Also I see we have some singletons for DT/TT features, I'm planning to remove those for many reasons (essentially it's an anti-pattern, and we don't need to define them as singleton)

### PR DESCRIPTION
**Pull request by Rapster**

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java [hunk 1]
```diff
@@ -77,7 +85,7 @@ protected void encodeScript(FacesContext context, ImageCropper cropper) throws I
         String select = null;
 
         WidgetBuilder wb = getWidgetBuilder(context);
-        wb.initWithComponentLoad("ImageCropper", widgetVar, clientId, clientId + "_image")
+        wb.initWithComponentLoad("ImageCropper", widgetVar, clientId, image)
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java:88]: are you sure with this? this is by design to attach the onload to the image witself, not the component propably

Comment (Rapster) [reply primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java:88]: I'm sure: `String image = clientId + "_image";` 😉

File: primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java [hunk 2]
```diff
@@ -284,50 +228,38 @@ else if (stream != null) {
      * returns the value obtained by ResourceHandler.createResource().
      */
     private Resource getImageResource(FacesContext facesContext, ImageCropper imageCropper) {
```
Comment (tandraschko) [inline primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java:230]: i wonder in general if we need this method?
we could just evaluate the EL, without splitting and so on?

Comment (Rapster) [reply primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java:230]: No idea, related to #1248 and I never worked with portlets

Comment (tandraschko) [reply primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java:230]: TBH i would revert that
#{resource[test:test]} must just work, is specified by JSF and resolve internally to the correct portlet

Comment (Rapster) [reply primefaces/src/main/java/org/primefaces/component/imagecropper/ImageCropperRenderer.java:230]: To discuss in another ticket then, here, the idea is just to clean up the code. But on a general note, I must admit it's kinda strange to see this only on ImageCropper, if it was really due to portlet environnement, I would picture that kinda change somewhere else

**Review by melloware — APPROVED** [review]
LGTM

**Pull request merged**